src/main/java/org/folio/service/CirculationSettingsService.java

```diff
@@ -54,17 +57,46 @@ public Future<Response> findById(String circulationSettingsId) {
       GetCirculationSettingsStorageCirculationSettingsByCirculationSettingsIdResponse.class);
   }
 
-  public Future<Response> update(String circulationSettingsId, CirculationSetting circulationSetting) {
-    return PgUtil.put(CIRCULATION_SETTINGS_TABLE, circulationSetting, circulationSettingsId, okapiHeaders, vertxContext,
+  public Future<Response> update(String circulationSettingsId,
+    CirculationSetting circulationSetting) {
+    return PgUtil.put(CIRCULATION_SETTINGS_TABLE, circulationSetting, circulationSettingsId,
+        okapiHeaders, vertxContext,
         PutCirculationSettingsStorageCirculationSettingsByCirculationSettingsIdResponse.class)
       .compose(eventPublisher.publishUpdated(circulationSetting));
   }
 
   public Future<Response> delete(String circulationSettingsId) {
-    return repository.getById(circulationSettingsId).compose (
-      circulationSetting -> PgUtil.deleteById(CIRCULATION_SETTINGS_TABLE, circulationSettingsId, okapiHeaders, vertxContext,
-        DeleteCirculationSettingsStorageCirculationSettingsByCirculationSettingsIdResponse.class)
-      .compose(eventPublisher.publishRemoved(circulationSetting))
+    return repository.getById(circulationSettingsId).compose(
+      circulationSetting -> PgUtil.deleteById(CIRCULATION_SETTINGS_TABLE, circulationSettingsId,
+          okapiHeaders, vertxContext,
+          DeleteCirculationSettingsStorageCirculationSettingsByCirculationSettingsIdResponse.class)
+        .compose(eventPublisher.publishRemoved(circulationSetting))
```

Comment:
`PgUtil.deleteById` will always return a succeeded future, even if deletion failed. This means that we will also publish an event if deletion failed.

Reply:
postponed